core/oracles/oracle_spec.go

```diff
@@ -71,70 +71,105 @@ type condition func(string) (bool, error)
 // NewOracleSpec builds an OracleSpec from a types.OracleSpec in a form that
 // suits the processing of the filters.
 // OracleSpec allows the existence of one and only one.
+// Currently VEGA network utilises internal triggers in the oracle function path, even though
+// the oracles are treated as external data sources.
+// For this reason this function checks if the provided external type of data source definition
+// contains a key name that indicates a builtin type of logic
+// and if the given data source definition is an internal type of data source, for more context refer to
+// https://github.com/vegaprotocol/specs/blob/master/protocol/0048-DSRI-data_source_internal.md#13-vega-time-changed
 func NewOracleSpec(originalSpec types.ExternalDataSourceSpec) (*OracleSpec, error) {
-	signersFromSpec := []*types.Signer{}
+	filtersFromSpec := []*types.DataSourceSpecFilter{}
+	isExtType := false
 	if originalSpec.Spec != nil {
 		if originalSpec.Spec.Data != nil {
-			src := *originalSpec.Spec.Data
-
-			signersFromSpec = src.GetSigners()
+			filtersFromSpec, isExtType = originalSpec.Spec.Data.GetFilters()
```

Comment:
ExternalDataSourceSpec contains the core type DataSourceSpec. Since we do not have other external sources at the moment, DataSourceSpec matches what we need for oracles - but this is intended to change in the future - then DataSourceSpec will contain only the common type of data for different data sources.
Oracles are handled as a type of external data sources at the moment, and that was in the intention. So NewOracleSpec receives an external data type.

But along with that, at the moment we still use the oracles to deliver data internally. So currently the oracle path handles this internal flow. I added this in one of the comments in the changed code as well.

Reply:
Do you mean the filters type to have a method showing if it is external or internal? There is no current indication for that inside the DataSourceSpecFilter object, it can be added but it will be a breaking change, not sure if big enough to cause issues though.
If you mean the DataSourceDefinition to have such method - it will make a check using the interface conversion and this is already done in GetFilters() and GetSigners(), so it will be done a third time. So in that case we will not use these methods, but directly interpret the type in the NewOracleSpec function. I think that will be better, will do it like this now.

Reply:
Ah ok I think I understand a little better now. Leave it as it is.